service/collector/env.go

```diff
@@ -41,19 +58,130 @@ func NewEnv(config EnvConfig) (*Env, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 
-	v := &Env{
+	if config.Directory == "" {
+		config.Directory = "/run/flannel/networks/"
+	}
+
+	c := &Env{
 		g8sClient: config.G8sClient,
 		logger:    config.Logger,
+
+		directory: config.Directory,
 	}
 
-	return v, nil
+	return c, nil
 }
 
-func (v *Env) Collect(ch chan<- prometheus.Metric) error {
+func (c *Env) Collect(ch chan<- prometheus.Metric) error {
+	var desiredClusterIDs []string
+	{
+		l, err := c.g8sClient.Core().FlannelConfigs(metav1.NamespaceAll).List(metav1.ListOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, i := range l.Items {
+			desiredClusterIDs = append(desiredClusterIDs, i.Name)
+		}
+	}
+
+	var envClusterIDs []string
+	{
+		f := func(path string, info os.FileInfo, err error) error {
+			envClusterIDs = append(envClusterIDs, clusterIDFromPath(path))
+			return nil
+		}
+
+		err := filepath.Walk(c.directory, f)
```

Comment:
I don't think it's a good use of `filepath .Walk` if we don't even check the file type. If you assume that only regular files are in the directory then `ioutil. ReadDir` is a better fit IMO.

Reply:
Shouldn't that just be fine when we do not configure the directory path? Like, what could be a problem here?

Reply:
If this is not a directory then `ioutl.ReadDir` should fail IIRC.

But that's probably a good thing? If this is not a directory then I think something is configured incorrectly.

Reply:
After some local testing I figured that `Walk` does not even fail on files that do not exist. I am going with your suggestion now.